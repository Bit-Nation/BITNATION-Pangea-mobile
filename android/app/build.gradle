apply plugin: "com.android.application"

//react native config
apply from: project(':react-native-config').projectDir.getPath() + "/dotenv.gradle"

import com.android.build.OutputFile

/**
 * The react.gradle file registers a task for each build variant (e.g. bundleDebugJsAndAssets
 * and bundleReleaseJsAndAssets).
 * These basically call `react-native bundle` with the correct arguments during the Android build
 * cycle. By default, bundleDebugJsAndAssets is skipped, as in debug/dev mode we prefer to load the
 * bundle directly from the development server. Below you can see all the possible configurations
 * and their defaults. If you decide to add a configuration block, make sure to add it before the
 * `apply from: "../../node_modules/react-native/react.gradle"` line.
 *
 * project.ext.react = [
 *   // the name of the generated asset file containing your JS bundle
 *   bundleAssetName: "index.android.bundle",
 *
 *   // the entry file for bundle generation
 *   entryFile: "index.android.js",
 *
 *   // whether to bundle JS and assets in debug mode
 *   bundleInDebug: false,
 *
 *   // whether to bundle JS and assets in release mode
 *   bundleInRelease: true,
 *
 *   // whether to bundle JS and assets in another build variant (if configured).
 *   // See http://tools.android.com/tech-docs/new-build-system/user-guide#TOC-Build-Variants
 *   // The configuration property can be in the following formats
 *   //         'bundleIn${productFlavor}${buildType}'
 *   //         'bundleIn${buildType}'
 *   // bundleInFreeDebug: true,
 *   // bundleInPaidRelease: true,
 *   // bundleInBeta: true,
 *
 *   // whether to disable dev mode in custom build variants (by default only disabled in release)
 *   // for example: to disable dev mode in the staging build type (if configured)
 *   devDisabledInStaging: true,
 *   // The configuration property can be in the following formats
 *   //         'devDisabledIn${productFlavor}${buildType}'
 *   //         'devDisabledIn${buildType}'
 *
 *   // the root of your project, i.e. where "package.json" lives
 *   root: "../../",
 *
 *   // where to put the JS bundle asset in debug mode
 *   jsBundleDirDebug: "$buildDir/intermediates/assets/debug",
 *
 *   // where to put the JS bundle asset in release mode
 *   jsBundleDirRelease: "$buildDir/intermediates/assets/release",
 *
 *   // where to put drawable resources / React Native assets, e.g. the ones you use via
 *   // require('./image.png')), in debug mode
 *   resourcesDirDebug: "$buildDir/intermediates/res/merged/debug",
 *
 *   // where to put drawable resources / React Native assets, e.g. the ones you use via
 *   // require('./image.png')), in release mode
 *   resourcesDirRelease: "$buildDir/intermediates/res/merged/release",
 *
 *   // by default the gradle tasks are skipped if none of the JS files or assets change; this means
 *   // that we don't look at files in android/ or ios/ to determine whether the tasks are up to
 *   // date; if you have any other folders that you want to ignore for performance reasons (gradle
 *   // indexes the entire tree), add them here. Alternatively, if you have JS files in android/
 *   // for example, you might want to remove it from here.
 *   inputExcludes: ["android/**", "ios/**"],
 *
 *   // override which node gets called and with what additional arguments
 *   nodeExecutableAndArgs: ["node"],
 *
 *   // supply additional arguments to the packager
 *   extraPackagerArgs: []
 * ]
 */

project.ext.react = [
        entryFile: "index.js",
        devDisabledInStaging: true
]

apply from: "../../node_modules/react-native/react.gradle"

/**
 * Set this to true to create two separate APKs instead of one:
 *   - An APK that only works on ARM devices
 *   - An APK that only works on x86 devices
 * The advantage is the size of the APK is reduced by about 4MB.
 * Upload all the APKs to the Play Store and people will download
 * the correct one based on the CPU architecture of their device.
 */
def enableSeparateBuildPerCPUArchitecture = false

/**
 * Run Proguard to shrink the Java bytecode in release builds.
 */
def enableProguardInReleaseBuilds = false

android {
    compileSdkVersion 28
    buildToolsVersion '28.0.3'
    defaultConfig {
        applicationId 'co.bitnation'
        minSdkVersion 16
        targetSdkVersion 22
        versionCode 17
        versionName "1.3.3"
        ndk {
            abiFilters "armeabi-v7a", "x86"
        }
        vectorDrawables.useSupportLibrary = true
        multiDexEnabled true

        // note: using ${applicationId} here will be exactly as above
        // and so NOT necessarily the applicationId of the generated APK
        resValue "string", "build_config_package", "${applicationId}"
    }
    signingConfigs {
        garbage_key {
            keyAlias 'androiddebugkey'
            keyPassword 'android'

            // use ~/.android/debug.keystore
            storeFile "${System.properties['user.home']}${File.separator}.android${File.separator}debug.keystore" as File
            storePassword 'android'
        }
        staging {
            if (project.hasProperty('MYAPP_STAGING_STORE_FILE')) {
                storeFile file(MYAPP_STAGING_STORE_FILE)
                storePassword MYAPP_STAGING_STORE_PASSWORD
                keyAlias MYAPP_STAGING_KEY_ALIAS
                keyPassword MYAPP_STAGING_KEY_PASSWORD
            }
        }
        release {
            if (project.hasProperty('MYAPP_RELEASE_STORE_FILE')) {
                storeFile file(MYAPP_RELEASE_STORE_FILE)
                storePassword MYAPP_RELEASE_STORE_PASSWORD
                keyAlias MYAPP_RELEASE_KEY_ALIAS
                keyPassword MYAPP_RELEASE_KEY_PASSWORD
            }
        }
    }
    splits {
        abi {
            reset()
            enable enableSeparateBuildPerCPUArchitecture
            universalApk false  // If true, also generate a universal APK
            include "armeabi-v7a", "x86"
        }
    }
    buildTypes {
        staging {
            initWith release
            applicationIdSuffix ".staging"
            debuggable false
            matchingFallbacks = ['release']
            signingConfig signingConfigs.staging
        }
        stagingLocal {
            initWith staging
            signingConfig signingConfigs.garbage_key
        }

        release {
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
            signingConfig signingConfigs.release
        }
    }
    // applicationVariants are e.g. debug, release
    applicationVariants.all { variant ->
        variant.outputs.each { output ->
            // For each separate APK per architecture, set a unique version code as described here:
            // http://tools.android.com/tech-docs/new-build-system/user-guide/apk-splits
            def versionCodes = ["armeabi-v7a": 1, "x86": 2]
            def abi = output.getFilter(OutputFile.ABI)
            if (abi != null) {  // null for the universal-debug, universal-release variants
                output.versionCodeOverride =
                        versionCodes.get(abi) * 1048576 + defaultConfig.versionCode
            }
        }
    }
    productFlavors {
    }
    lintOptions {
        checkReleaseBuilds false
    }
    dexOptions {
        jumboMode true
    }
}

dependencies {
    api project(':react-native-shake')
    api (project(':react-native-camera')) {
      exclude group: "com.android.support", module: 'support-v4'
    }
    api project(':react-native-image-crop-picker')
    api project(':realm')
    api project(':react-native-vector-icons')
    api project(':react-native-sensitive-info')
    api project(':react-native-randombytes')
    api project(':react-native-navigation')
    api project(':react-native-config')
    api project(':react-native-snackbar')
    implementation project(':panthalassa')
    testCompile 'junit:junit:4.12'
    testCompile 'org.mockito:mockito-core:1.10.19'
    api fileTree(include: ['*.jar'], dir: 'libs')
    api 'com.android.support:appcompat-v7:25.0.1'
    implementation 'com.facebook.react:react-native:+'
}


// Run this once to be able to run the application with BUCK
// puts all compile dependencies into folder libs for BUCK to use
task copyDownloadableDepsToLibs(type: Copy) {
    from configurations.api
    into 'libs'
}
