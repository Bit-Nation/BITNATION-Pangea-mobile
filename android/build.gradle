// Top-level build file where you can add configuration options common to all sub-projects/modules.


buildscript {
    repositories {
        google()
        jcenter()
 
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:3.0.1'
    }
}

allprojects {
    repositories {
        mavenLocal()
        maven { url "https://jitpack.io" }
        maven { url "https://maven.google.com" }
        google()
        jcenter()
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url "$rootDir/../node_modules/react-native/android"
        }
        maven { url "https://jitpack.io" }
        flatDir{
            dirs "$rootDir/panthalassa"
        }
    }
}