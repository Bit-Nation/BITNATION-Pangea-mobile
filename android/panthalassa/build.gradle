configurations.maybeCreate("default")
artifacts.add("default", file('panthalassa.aar'))