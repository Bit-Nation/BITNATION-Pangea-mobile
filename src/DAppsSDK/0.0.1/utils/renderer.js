// @flow

import React from 'react';
import components from '../components';

export type ComponentsJSON = { type: string, props: Object, children: Array<Object> };

/**
 * @desc Function to get element class or function from text representation, e.g. 'Text' means Text component.
 * @param {string} typeName String representation of the type.
 * @return {*} Class or function of component or null if type is unknown.
 */
export const getTypeElementFromText = (typeName: string) => components[typeName];

/**
 * @desc Validates props against list of valid names and filter only whitelisted ones.
 * Used to prevent DApp developer to pass something bad to components we provide.
 * @param {Object} props Props to validate.
 * @param {string[]} validProps Object containing arrays by keys `native`, `stateBased` and `callbacks` according to corresponding type of props.
 * @param {string} type String representation of component type.
 * @return {Object} Props filtered by groups.
 */
export const validateProps = (
  props: Object,
  validProps: { native: Array<string>, custom: Array<string>, callbacks: Array<string> },
  type: string,
) => {
  const nativeProps = {};
  const customProps = {};
  const callbackProps = {};
  Object.keys(props).forEach((propName) => {
    if (typeof propName !== 'string') {
      console.warn('Prop name is not a string');
      return;
    }

    if (validProps.native.includes(propName)) {
      // It's an allowed native prop.
      nativeProps[propName] = props[propName];
    } else if (validProps.custom.includes(propName)) {
      // It's an allowed custom prop.
      customProps[propName] = props[propName];
    } else if (propName.endsWith('ID') && validProps.callbacks.includes(propName.substring(0, propName.length - 2))) {
      // It's an allowed prop to perform a callback. That properties are marked by ID suffix.
      callbackProps[propName] = props[propName];
    } else {
      console.warn(`Prop ${propName} is not allowed on component ${type}`);
    }
  });

  return { nativeProps, customProps, callbackProps };
};

/**
 * @desc Renders JSON provided by DApp developer into react components.
 * @param {ComponentsJSON} json JSON object that represents components to render.
 * @param {?string} key Key of component (used for updates).
 * @param {Function} customPropsProvider Function that takes component class or function and props and returns new props if it's needed to add ones.
 * @param {*} parent Function or class of component that is parent of currently used or undefined for root.
 * @return {*} Tree of components that is ready to be render.
 */
export const renderJSON = (json: ComponentsJSON, key: ?string, customPropsProvider: (component: any, ownProps: Object) => Object, parent: any) => {
  const { type, children } = json;

  // This is the case when we have string literal inside children.
  // It's only allowed for Text.
  if (type === undefined && typeof (json) === 'string') {
    if (parent === components.Text) {
      return json;
    }

    console.warn(`String literal may only appear inside Text component. Found "${json}" outside.`);
    return null;
  }

  const component = components[type];

  if (component == null) {
    console.warn(`Trying to render unknown component type ${type}.`);
    return null;
  }

  if (component.validNativeProps != null && Array.isArray(component.validNativeProps) === false) {
    console.warn(`validNativeProps field of component ${type} should be an array of strings.`);
    return null;
  }
  if (component.customProps != null && Array.isArray(component.customProps) === false) {
    console.warn(`customProps field of component ${type} should be an array of strings.`);
    return null;
  }
  if (component.callbackProps != null && Array.isArray(component.callbackProps) === false) {
    console.warn(`callbackProps field of component ${type} should be an array of strings.`);
    return null;
  }

  const { props } = json;

  const { nativeProps = props, customProps = {}, callbackProps = {} } =
    validateProps(json.props, {
      native: component.validNativeProps || [],
      custom: component.customProps || [],
      callbacks: component.callbackProps || [],
    }, type);

  return React.createElement(
    component,
    {
      nativeProps,
      ...customProps,
      ...customPropsProvider(component, { ...nativeProps, ...customProps, ...callbackProps }),
      key,
    },
    children
      ? children.map((child, index) => renderJSON(child, `${index}`, customPropsProvider, component))
      : null,
  );
};
