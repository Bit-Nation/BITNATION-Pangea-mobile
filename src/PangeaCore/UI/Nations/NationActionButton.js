// @flow

import React from 'react';
import {
  Image,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';

import { MediaQueryStyleSheet } from 'react-native-responsive';
import GlobalStyles from 'pangea-common-reactnative/styles';

type Props = {
  /**
   * @desc Value to enable/disable the button.
   */
  disable?: boolean,
  /**
   * @desc Text label of the button.
   */
  title?: string,
  /**
   * @desc Callback to be called when button is pressed.
   */
  onPress?: () => void,
  /**
   * @desc Resource to be rendered as icon with the button.
   */
  iconSource?: number,
}

/**
 * @desc Component that renders the buttons used on the Nations Screens (icon + text)
 * @return {React.Component} A component.
 */

const NationActionButton = ({
  title, disable, iconSource, onPress,
}: Props) => {
  const styles = MediaQueryStyleSheet.create({
    ...GlobalStyles,
  });

  return (
    <View style={styles.tabBarButton} opacity={disable ? 0.4 : 1}>
      <TouchableOpacity
        style={[styles.tabBarContainer]}
        onPress={onPress}
        disabled={disable}
        activeOpacity={disable ? 1 : 0.4}
      >
        <Image source={iconSource} />
        <Text style={styles.tabBarTitle}>{title}</Text>
      </TouchableOpacity>
    </View>
  );
};

NationActionButton.defaultProps = {
  disable: true,
  title: '',
  iconSource: undefined,
  onPress: () => null,
};

export default NationActionButton;
