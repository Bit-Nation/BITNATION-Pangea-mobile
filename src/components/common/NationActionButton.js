import React, { Component } from 'react';
import { Image, Text, TouchableOpacity, View } from 'react-native';
import { MediaQueryStyleSheet } from 'react-native-responsive';

import Colors from '../../global/Colors';
import GlobalStyles from '../../global/Styles';

{/*  *************************
  NationActionButton parameters:
    iconSource = image to render
    title = Title of the Button
    disable = State of the Button
    ****************************  */}

export default class NationActionButton extends Component {

  render() {
    const { style, children, ...props } = this.props;

    return (
      <View style={[styles.tabBarButton, style]} {...props} opacity={this.props.disable ? 0.4 : 1}>
        <TouchableOpacity style={[styles.tabBarContainer]}
                          onPress={this.props.onPress} disabled={this.props.disable}
                          activeOpacity={this.props.disable ? 1 : 0.4}>
          <Image source={this.props.iconSource}/>
          <Text style={styles.tabBarTitle}>{this.props.title}</Text>
        </TouchableOpacity>
      </View>
    );
  }

}

const styles = MediaQueryStyleSheet.create({
  ...GlobalStyles,

});