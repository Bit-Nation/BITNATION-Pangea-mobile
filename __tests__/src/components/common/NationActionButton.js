import NationActionButton from '../../../../src/components/common/NationActionButton';
import React from 'react';
import renderer from 'react-test-renderer'
import AssetsImage from "../../../../src/global/AssetsImages";
import {shallow} from "enzyme/build/index";
import {TouchableOpacity} from "react-native";

describe('NationActionButton component tests', () => {

  test('NationActionButton renders correctly', () => {
    const tree = renderer.create(<NationActionButton
      iconSource={AssetsImage.Actions.chat}
      title='Chat' disable={true}/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('NationActionButton press works', () => {
    const onPress = jest.fn();
    const wrapper = shallow(
      <NationActionButton onPress={onPress} enabled={true}/>
    );
    expect(wrapper).toMatchSnapshot();
    const render = wrapper.dive();
    render.find(TouchableOpacity).simulate('press');
    expect(onPress).toHaveBeenCalledTimes(1);
  });

  test('NationActionButton press disabled', () => {
    const onPress = jest.fn();
    const wrapper = shallow(
      <NationActionButton onPress={onPress} disable={true}/>
    );
    expect(wrapper).toMatchSnapshot();
    const render = wrapper.dive();
    render.find(TouchableOpacity).simulate('press');
    expect(render.props().disable).toBeTruthy();
  });

  test('NationActionButton press enabled', () => {
    const onPress = jest.fn();
    const wrapper = shallow(
      <NationActionButton
        onPress={onPress}
        disable={false}
        iconSource={AssetsImage.Actions.chat}
        title='TEST'
      />
    );
    expect(wrapper).toMatchSnapshot();
    const render = wrapper.dive();
    render.find(TouchableOpacity).simulate('press');
    expect(render.props().disable).toBeFalsy();
  });

});