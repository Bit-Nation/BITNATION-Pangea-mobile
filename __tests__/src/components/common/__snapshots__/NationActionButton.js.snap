// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NationActionButton component tests NationActionButton press disabled 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NationActionButton
    disable={true}
    iconSource={undefined}
    onPress={[MockFunction]}
    title=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <TouchableOpacity
        activeOpacity={1}
        disabled={true}
        onPress={[MockFunction]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <Image
          source={undefined}
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "white",
              "fontSize": 10,
              "marginTop": 4,
            }
          }
        >
          
        </Text>
      </TouchableOpacity>,
      "opacity": 0.4,
      "style": Object {
        "backgroundColor": "transparent",
        "borderRadius": 15,
        "height": 48,
        "justifyContent": "center",
        "width": 48,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 1,
        "children": Array [
          <Image
            source={undefined}
          />,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 10,
                "marginTop": 4,
              }
            }
          >
            
          </Text>,
        ],
        "disabled": true,
        "onPress": [MockFunction],
        "style": Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "source": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "",
            "ellipsizeMode": "tail",
            "style": Object {
              "backgroundColor": "transparent",
              "color": "white",
              "fontSize": 10,
              "marginTop": 4,
            },
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <TouchableOpacity
          activeOpacity={1}
          disabled={true}
          onPress={[MockFunction]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Image
            source={undefined}
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 10,
                "marginTop": 4,
              }
            }
          >
            
          </Text>
        </TouchableOpacity>,
        "opacity": 0.4,
        "style": Object {
          "backgroundColor": "transparent",
          "borderRadius": 15,
          "height": 48,
          "justifyContent": "center",
          "width": 48,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 1,
          "children": Array [
            <Image
              source={undefined}
            />,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 10,
                  "marginTop": 4,
                }
              }
            >
              
            </Text>,
          ],
          "disabled": true,
          "onPress": [MockFunction],
          "style": Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "",
              "ellipsizeMode": "tail",
              "style": Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 10,
                "marginTop": 4,
              },
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`NationActionButton component tests NationActionButton press enabled 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NationActionButton
    disable={false}
    iconSource={1}
    onPress={[MockFunction]}
    title="TEST"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <TouchableOpacity
        activeOpacity={0.4}
        disabled={false}
        onPress={[MockFunction]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <Image
          source={1}
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "white",
              "fontSize": 10,
              "marginTop": 4,
            }
          }
        >
          TEST
        </Text>
      </TouchableOpacity>,
      "opacity": 1,
      "style": Object {
        "backgroundColor": "transparent",
        "borderRadius": 15,
        "height": 48,
        "justifyContent": "center",
        "width": 48,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.4,
        "children": Array [
          <Image
            source={1}
          />,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 10,
                "marginTop": 4,
              }
            }
          >
            TEST
          </Text>,
        ],
        "disabled": false,
        "onPress": [MockFunction],
        "style": Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "source": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "TEST",
            "ellipsizeMode": "tail",
            "style": Object {
              "backgroundColor": "transparent",
              "color": "white",
              "fontSize": 10,
              "marginTop": 4,
            },
          },
          "ref": null,
          "rendered": "TEST",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <TouchableOpacity
          activeOpacity={0.4}
          disabled={false}
          onPress={[MockFunction]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Image
            source={1}
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 10,
                "marginTop": 4,
              }
            }
          >
            TEST
          </Text>
        </TouchableOpacity>,
        "opacity": 1,
        "style": Object {
          "backgroundColor": "transparent",
          "borderRadius": 15,
          "height": 48,
          "justifyContent": "center",
          "width": 48,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.4,
          "children": Array [
            <Image
              source={1}
            />,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 10,
                  "marginTop": 4,
                }
              }
            >
              TEST
            </Text>,
          ],
          "disabled": false,
          "onPress": [MockFunction],
          "style": Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "TEST",
              "ellipsizeMode": "tail",
              "style": Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 10,
                "marginTop": 4,
              },
            },
            "ref": null,
            "rendered": "TEST",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`NationActionButton component tests NationActionButton press works 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NationActionButton
    disable={true}
    enabled={true}
    iconSource={1}
    onPress={[MockFunction]}
    title=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <TouchableOpacity
        activeOpacity={1}
        disabled={true}
        onPress={[MockFunction]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <Image
          source={1}
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "white",
              "fontSize": 10,
              "marginTop": 4,
            }
          }
        >
          
        </Text>
      </TouchableOpacity>,
      "opacity": 0.4,
      "style": Object {
        "backgroundColor": "transparent",
        "borderRadius": 15,
        "height": 48,
        "justifyContent": "center",
        "width": 48,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 1,
        "children": Array [
          <Image
            source={1}
          />,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 10,
                "marginTop": 4,
              }
            }
          >
            
          </Text>,
        ],
        "disabled": true,
        "onPress": [MockFunction],
        "style": Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "source": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "",
            "ellipsizeMode": "tail",
            "style": Object {
              "backgroundColor": "transparent",
              "color": "white",
              "fontSize": 10,
              "marginTop": 4,
            },
          },
          "ref": null,
          "rendered": "",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <TouchableOpacity
          activeOpacity={1}
          disabled={true}
          onPress={[MockFunction]}
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
            ]
          }
        >
          <Image
            source={1}
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 10,
                "marginTop": 4,
              }
            }
          >
            
          </Text>
        </TouchableOpacity>,
        "opacity": 0.4,
        "style": Object {
          "backgroundColor": "transparent",
          "borderRadius": 15,
          "height": 48,
          "justifyContent": "center",
          "width": 48,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 1,
          "children": Array [
            <Image
              source={1}
            />,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 10,
                  "marginTop": 4,
                }
              }
            >
              
            </Text>,
          ],
          "disabled": true,
          "onPress": [MockFunction],
          "style": Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "",
              "ellipsizeMode": "tail",
              "style": Object {
                "backgroundColor": "transparent",
                "color": "white",
                "fontSize": 10,
                "marginTop": 4,
              },
            },
            "ref": null,
            "rendered": "",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`NationActionButton component tests NationActionButton renders correctly 1`] = `
<View
  opacity={0.4}
  style={
    Object {
      "backgroundColor": "transparent",
      "borderRadius": 15,
      "height": 48,
      "justifyContent": "center",
      "width": 48,
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Image
      source={1}
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "white",
          "fontSize": 10,
          "marginTop": 4,
        }
      }
    >
      Chat
    </Text>
  </View>
</View>
`;
